### ISSUE TITLE
Fix reduce type definition in prelude

### ISSUE THREAD

**Issue by Frikki** [TylorS/typed#35]
The type definition for `reduce()` in _@typed/prelude_ does not match the type definition for `reduce()` in _@typed/list_.

**@typed/list:**

```ts
export declare type Reducer<A, B> = (seed: B, value: A, index: number) => B;
export declare type ReduceArity3 = {
    <A, B>(f: Reducer<A, B>, seed: B, list: List<A>): B;
    <A, B>(f: Reducer<A, B>, seed: B): ReduceArity1<A, B>;
    <A, B>(f: Reducer<A, B>): ReduceArity2<A, B>;
};
export declare type ReduceArity2<A, B> = {
    (seed: B, list: List<A>): B;
    (seed: B): ReduceArity1<A, B>;
};
export declare type ReduceArity1<A, B> = {
    (list: List<A>): B;
};
```

**@typed/prelude:**

```ts
export function reduce<A, B>(f: (acc: A, val: B) => A, seed: A, arr: B[]): A
```

The “prelude”-reduce doesn’t allow `ReadonlyArray`s as lists, for example.

**Comment by TylorS** [comment]
It may not be obvious that it supports `ReadonlyArray`, also `Array` and `ArrayLike`, because they are valid implementations of `List`.

Beyond that, it seems you'd like to have the name of the first parameter to `Reducer<A, B>` be renamed to `accumulator`? I'd be happy to see that change 👍

**Comment by Frikki** [comment]
The `prelude` export is not `List<A>` but `B[]`. Thus, you _can not_ use a `ReadonlyArray` nor `ArrayLike` if you operate with the `prelude` version. In addition, `prelude`’s `reduce` specifies the first template type `A` as the `seed` and return type, whereas the `list` version has the template types reversed.

**Comment by TylorS** [comment]
Not to dismiss the issue you're running into, but I've been unable to recreate your issue.
<img width="581" alt="screen shot 2017-11-20 at 9 19 55 am" src="https://user-images.githubusercontent.com/6282473/33022728-1a0d627c-cdd4-11e7-92c6-3cb3a654179f.png">

**Comment by Frikki** [comment]
You’re right, you’re right. The code I was looking at was importing `reduce` from `@most/prelude`. Must have been tired. My bad.

I still would like to introduce the parameter name change, though.
